src/align_stack.cpp

```diff
@@ -236,8 +235,7 @@ void AlignStack::Add(chunk_t *start, int seqnum)
       {
          tmp = chunk_get_next(tmp);
       }
-      if ((chunk_is_star(tmp) && (m_star_style == SS_DANGLE)) ||
-          (chunk_is_addr(tmp) && (m_amp_style == SS_DANGLE)))
+      if (chunk_is_ptr_operator(tmp) && (m_star_style == SS_DANGLE))
```

Comment:
This looks like it changes the logic.  m_star_style and m_amp_style are different.
You likely want '^' to follow m_start_style.  Same for the chunk below.